llmlite/apis/chatllm.py

```diff
@@ -79,6 +81,11 @@ def completion(
         Returns:
             Sentences of string type.
         """
+        
+        if type(messages[0]) != list:
+            messages=[messages]
+        else:
+            assert self.used_backend == consts.BACKEND_VLLM, "the backend does not support batching"
```

Comment:
What we should do here is 
```
if backend != vllm and messages is batch:
    raise Exception
```

We already have the self.backend property

Reply:
So generally, if the messages is batch, we should only go through vllm.

Reply:
I understand what you mean, but what i want to do is that process message to List[List[ChatMessage]] class to unify subsequent inputs of completion.

Reply:
sgtm, however, if you want to add validations, you should add it as early as possible, that's say you should add the validation in the `chatllm.competition()`.

Reply:
rename to self.backend, that's enough.